crates/bevy_render/src/camera/camera.rs

```diff
@@ -343,19 +347,47 @@ impl Camera {
     /// To get the world space coordinates with the viewport position, you should use
     /// [`world_to_viewport`](Self::world_to_viewport).
     ///
-    /// Returns `None` if the `camera_transform`, the `world_position`, or the projection matrix defined by [`CameraProjection`] contain `NAN`.
     /// Panics if the projection matrix is null and `glam_assert` is enabled.
-    pub fn ndc_to_world(&self, camera_transform: &GlobalTransform, ndc: Vec3) -> Option<Vec3> {
+    pub fn ndc_to_world(
+        &self,
+        camera_transform: &GlobalTransform,
+        ndc: Vec3,
+    ) -> Result<Vec3, NdcToWorldError> {
+        let cam_trans_mat = camera_transform.compute_matrix();
+        if cam_trans_mat.is_nan() {
+            return Err(NdcToWorldError::CameraTransformNanError(cam_trans_mat));
+        }
+
+        let proj_mat = self.computed.projection_matrix;
+        if proj_mat.is_nan() {
+            return Err(NdcToWorldError::ProjectionMatrixNanError(proj_mat));
+        }
+
         // Build a transformation matrix to convert from NDC to world space using camera data
-        let ndc_to_world =
-            camera_transform.compute_matrix() * self.computed.projection_matrix.inverse();
+        let ndc_to_world = cam_trans_mat * proj_mat.inverse();
```

Comment:
I'd personally prefer if the variables used the full words here. Like `camera_transform_matrix` and `projection_matrix`. But that's just my preference I don't consider this a blocker.

Reply:
> I'd personally prefer if the variables used the full words here. Like `camera_transform_matrix` and `projection_matrix`. But that's just my preference I don't consider this a blocker.

Especially as `mat` can easily be read as `mut`, which is also not unlikely to read in Rust